.github/workflows/build-sola.yml

```diff
@@ -0,0 +1,164 @@
+      with:
+        path: ~/ns-3_install
+        key: ns-3_install-${{env.NS_3_VERSION}}
+
+  clangtidy:
+    needs: ns-3_compile
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+    - name: Download ns-3 cache
+      uses: actions/cache@v3
+      with:
+        path: ~/ns-3_install
+        key: ns-3_install-${{env.NS_3_VERSION}}
+        fail-on-cache-miss: true
+    - name: Install dependencies
+      run: sudo apt-get install -y libyaml-cpp-dev libevent-dev libsqlite3-dev
+    - name: Configure build
+      run: |
+        mkdir build && cd build
+        cmake ../daisi -DCMAKE_BUILD_TYPE=RelWithDebInfo -Dns3_DIR=~/ns-3_install/lib/cmake/ns3 -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+        cd ..
+        python3 scripts/compile_commands_clean.py build/compile_commands.json third_party tests
+    - name: Run clang-tidy
+      run: |
+        run-clang-tidy -header-filter "(daisi|minhton|natter|sola|solanet)/(src|include|examples).*" -p build
+
+  cppcheck:
+    needs: ns-3_compile
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+    - name: Download ns-3 cache
+      uses: actions/cache@v3
+      with:
+        path: ~/ns-3_install
+        key: ns-3_install-${{env.NS_3_VERSION}}
+        fail-on-cache-miss: true
+    - name: Install dependencies
+      run: sudo apt-get install -y libyaml-cpp-dev libevent-dev libsqlite3-dev cppcheck
+    - name: Configure build
+      run: |
+        mkdir build && cd build
+        cmake ../daisi -DCMAKE_BUILD_TYPE=RelWithDebInfo -Dns3_DIR=~/ns-3_install/lib/cmake/ns3 -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+        cd ..
+        python3 scripts/compile_commands_clean.py build/compile_commands.json third_party
+    - name: Run cppcheck
+      run: |
+        cppcheck --enable=all --error-exitcode=2 --std=c++17 --project=build/compile_commands.json -DCPPCHECK_IGNORE -j `nproc` --suppressions-list=cppcheck.conf
+
+  build_and_check:
+    needs: ns-3_compile
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+    - name: Download ns-3 cache
+      uses: actions/cache@v3
+      with:
+        path: ~/ns-3_install
+        key: ns-3_install-${{env.NS_3_VERSION}}
+        fail-on-cache-miss: true
+    - name: Install dependencies
+      run: sudo apt-get install -y libyaml-cpp-dev libevent-dev libsqlite3-dev cppcheck ninja-build
+    - name: Build
+      run: |
+        mkdir build && cd build
+        cmake ../daisi -G Ninja -DCMAKE_BUILD_TYPE=Release -Dns3_DIR=~/ns-3_install/lib/cmake/ns3 -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+        cmake --build .
+    - name: Run MINHTON unittest
+      run: build/sola_mono/minhton/tests/MinhtonAllTests
+    - name: Run Natter unittest
+      run: build/sola_mono/natter/tests/NatterAllTests
+    - name: Run CPPS unittest
+      run: build/tests/unittests/CPPSAmrMobilityHelperTest
```

Comment:
add running unittests for...
DaisiDatastructureDirectedGraphTest
DaisiDatastructureWeightedDirectedGraphTest
DaisiDatastructureSimpleTemporalNetworkTest
DaisiCppsOrderManagementStnOrderManagement

or should we add something like CppsAllTests?

Reply:
For now, I added all **DAISI** unittests individually.